sdk/src/main/java/co/decodable/sdk/pipeline/DecodableDataStreamSourceBuilder.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright Decodable, Inc.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package co.decodable.sdk.pipeline;
+
+import co.decodable.sdk.pipeline.util.Incubating;
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+@Incubating
+public class DecodableDataStreamSourceBuilder<T> {
```

Comment:
Why do we have these new builder classes? There already is `DecodableStreamSourceBuilder` and the internal implementation `DecodableStreamSourceBuilderImpl` (same for sink).

Reply:
These builders construct a different set of objects than the existing `DecodableStreamSourceBuilder` / `DecodableStreamSinkBuilder`, building `DataStreamSource` / `DataStreamSink` rather than `DecodableStreamSource` / `DecodableStreamSink` (which are `StatefulSource` / `StatefulSink` respectively). 

Why? Because we can't enforce an operator naming convention at the `StatefulSource` level, but we can at the `DataStreamSource` level.

Reply:
changed https://github.com/decodableco/decodable-pipeline-sdk/pull/41/commits/fa4608a93d960b19f881afb13f41a906090d0d3f